src/pool/srv_pool.c

```diff
@@ -2976,14 +2976,15 @@ ds_pool_check_leader(uuid_t pool_uuid, daos_unit_oid_t *oid,
 	if (rc != 0)
 		goto out;
 
-	leader = pl_select_leader(oid->id_pub, oid->id_shard, layout->ol_nr,
+	rc = pl_select_leader(oid->id_pub, oid->id_shard, layout->ol_nr,
 				  true, ds_pool_get_shard, layout);
-	if (leader < 0) {
+	if (rc < 0) {
 		D_WARN("Failed to select leader for "DF_UOID
 		       "version = %d: rc = %d\n",
-		       DP_UOID(*oid), version, leader);
-		D_GOTO(out, rc = leader);
+		       DP_UOID(*oid), version, rc);
+		D_GOTO(out, rc);
 	}
+	leader = rc;
```

Comment:
There is another dedicated patch for that:
https://github.com/daos-stack/daos/pull/272/files

Reply:
Ah, ok, i'll remove my change for this then.